### ISSUE TITLE
colormap zcol value has wrong length when z has missing values

### ISSUE THREAD

**Issue by richardsc** [dankelley/oce#1082]
Best to illustrate with an example. Basically what is happening is that any values in `z` that are NA are dropped when constructing the `zcol` return value. If `zcol` is then used to color points in a plot, the colours will not match the original data. Note that because R will recycle short vectors, it doesn't kick up a fuss. I've had a dig around in the `colors.R` function to see where this is going wrong, but haven't found the obvious candidate yet.
```r
library(oce)
z <- rnorm(1000)
y <- z
y[y>1] <- NA
zcm <- colormap(z)
ycm <- colormap(y)
par(mfrow=c(1, 2))
plot(z, col=zcm$zcol)
plot(y, col=ycm$zcol)
```
![image](https://cloud.githubusercontent.com/assets/233584/18640124/28f17394-7e6d-11e6-9ab4-e3af13b2f2a2.png)

**Comment by richardsc** [comment]
Note the length of `zcol` in the return -- `zcol` has length 838 instead of the required 1000:
```r
> str(ycm)
List of 10
 $ zlim        : num [1:2] -3.72 1.18
 $ breaks      : num [1:200] -3.72 -3.7 -3.67 -3.65 -3.63 ...
 $ col         : chr [1:199] "#00007F" "#000084" "#000089" "#00008E" ...
 $ zcol        : chr [1:838] "#FFBF00" "#FF3400" "#FF5800" "#FF4900" ...
 $ missingColor: chr "gray"
 $ x0          : num [1:199] -3.7 -3.67 -3.65 -3.63 -3.6 ...
 $ x1          : num [1:199] -3.7 -3.67 -3.65 -3.63 -3.6 ...
 $ col0        : chr [1:199] "#00007F" "#000084" "#000089" "#00008E" ...
 $ col1        : chr [1:199] "#00007F" "#000084" "#000089" "#00008E" ...
 $ zclip       : logi FALSE
 - attr(*, "class")= chr [1:2] "list" "colormap"
```

**Comment by richardsc** [comment]
Ok, I'm guessing that the issue is coming from this commit:

https://github.com/dankelley/oce/commit/8fb5ca72896a9ac0ec5f45c3703802335b73b580

which was clearly a fix for a different issue ...

**Comment by richardsc** [comment]
Ok, looks like it was done in the context of Issue #1000:

https://github.com/dankelley/oce/issues/1000#issuecomment-229684358

> @dankelley wrote: I just tweaked develop in commit 8fb5ca7 to look only at finite z values. Seem okay, Clark?

From a quick read of the issue, it seems like this change was made so that the z *limits* didn't come out with NAs. But that doesn't help the actual values when you want to plot as points. I'm pretty sure that things worked before that change, so it might be a good idea to try re-doing that fix (so the `zlim`s are finite) without changing the input z (which might have NAs)

**Comment by richardsc** [comment]
See PR #1083; I think it fixes this issue, without messing up the original reason for the change.

**Comment by richardsc** [comment]
This appears to be fixed via PR #1083, so I'm closing.
